content/blog/microcks-realm-fix-for-apicurio.md

```diff
@@ -0,0 +1,142 @@
+* Here in the Client roles drop-down select *microcks-app.* You should see the an entry called "*user*" in Assigned Roles and Effective Roles sections (both should have it)
+* Once this confirmed. Navigate to Clients->*microcks-app-js*
+* Here, click on the Client scopes and then Evaluate
+* In the user input, enter the name of the user whom you want to check and then click "Evaluate"
+
+![keycloak-client-scope-evaluation.png](/home/narioinc89/Downloads/images/blog/keycloak-client-scope-evaluation.png.png)
+
+* In the form that pops up up below, click on "Effective Role Scope Mapping"
+* Here, under Client roles drop down, select *microcks-app*.
+* You should see the "*user*" role in the Granted Effective Client Role section.
+
+#### Logging in to Microcks !
+
+Now login to Microcks app and mock away !!! You should see any APIs that you posted from APICurio or specs that you manually uploaded using the Importers section, come up correctly in the *Dashboard* and the *APIs | Services* section
+
+#### Skipping TLS !!!!
+
+**Dragons ahead !! BEWARE**, at no point should this be done for production environment. TLS is one of the first steps to ensuring a strong and secure environment for the tools we are working with and at no point should you disable SSL-required configuration in Keycloak. This is ONLY for development purposes. Also this change is ONLY required when you make external calls to these services. If you provide your APICurio IP in the range of *192.168.\*.\** or *127.0.0.1* or *localhost* you can just set the SSL-requireed setting in your Keycloak realm's 'Login' setting to false (or OFF as in the UI console). 
+
+For those independent developers and coders/hobbyist, use services like Let's Encrypt to get your free TLS certs to use with these services.  
+
+Now for the steps:
+
+* Stop your docker-compose installation of APICurio. 
+
+  ```
+  docker-compose -f "<name of the compose files>" down
+  ```
+
+* Download the application.properties file into you APICurio's docker-compose config folder as shown below
+
+  ```
+  cd docker-compose/config
+  
+  wget https://raw.githubusercontent.com/microcks/microcks/master/install/docker-compose/config/application.properties
+  
+  wget https://raw.githubusercontent.com/microcks/microcks/master/install/docker-compose/config/logback.xml
+  ```
+
+  
+
+* Comment 'all' the lines in application.properties except the following lines and change them as shown below:
+
+  ```
+  security.require-ssl=false
+  
+  keycloak.ssl-required=false
+  ```
+
+* Restart your APICurio installation. The Keycloak system and the RequestAuthenticator class will not complain of SSL-required for external requests now. You can check in the docker logs as well. 
+
+
+
+I hope this blog post provides an interim solution to the APICurio-Microcks installation woes. I will be sending out a PR for the permanent fix for this to the APICurio team. Until then, Happy Mocking !!
```

Comment:
Please add a link to your PR as well ;-)

Reply:
@lbroudoux  I did request the APICurio team if I could push a PR for the bug i raised but haven't received the confirmation on the participation. I will change the blog post to point to my bug report which was accepted and is beign worked upon

Reply:
Changed content to a reflect a bug instead of a PR. Added the link to the bug report